pci/src/bus.rs

```diff
@@ -356,8 +356,32 @@ impl BusDevice for PciConfigMmio {
     }
 }
 
+fn shift_and_mask(value: u32, offset: usize, mask: u32) -> usize {
+    ((value >> offset) & mask) as usize
+}
+
+// Parse the MMIO address offset to a (bus, device, function, register) tuple.
+// See https://wiki.osdev.org/PCI_Express#Enhanced_Configuration_Mechanism
```

Comment:
Ok so now everything makes sense cause I looked at the proper spec, where I think all the information are contained. It'd be more appropriate to refer to the spec rather than `wiki.osdev.org`.
I found the details about ECAM in section `7.2.2 PCI Express Enhanced Configuration Access Mechanism (ECAM)` from the `Pci Express Base Specification Revision 5.0 Version 1.0`.

Reply:
It's done already.